src/ast/irbuilderbpf.cpp

```diff
@@ -1337,5 +1346,25 @@ StoreInst *IRBuilderBPF::createAlignedStore(Value *val,
 #endif
 }
 
+llvm::Type *IRBuilderBPF::getPointerStorageTy()
+{
+  switch (sizeof(uintptr_t))
```

Comment:
Good catch! The case where bpftrace runs in compat mode on a 64-bit kernel isn't something I've considered, but it could be useful to support it -- or at least, as you noted, log an error and abort rather than producing wrong results.